test/reputation-system/happy-paths.js

```diff
@@ -951,12 +951,59 @@ contract("Reputation Mining - happy paths", (accounts) => {
 
       repCycle = await getActiveRepCycle(colonyNetwork);
       await forwardTime(SUBMITTER_ONLY_WINDOW + 1, this);
-      await repCycle.confirmNewHash(1);
+      await repCycle.confirmNewHash(1, { from: MINER1 });
 
       // Check it 'decayed' from 0 to 0
       expect(
         "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002"
       ).to.equal(goodClient.reputations[userKey]);
     });
+
+    it("a miner using delegated mining should be able to go through the whole process", async () => {
+      const badClient = new MaliciousReputationMinerExtraRep({ loader, realProviderPort, useJsTree, minerAddress: MINER2 }, 1, 0xfffffffff);
+      await badClient.initialise(colonyNetwork.address);
+      const delegatedClient = new ReputationMinerTestWrapper({ loader, realProviderPort, useJsTree, minerAddress: WORKER });
+      await colonyNetwork.setMiningDelegate(WORKER, true, { from: MINER1 });
+      await delegatedClient.initialise(colonyNetwork.address);
+      await advanceMiningCycleNoContest({ colonyNetwork, test: this });
+
+      const addr = await colonyNetwork.getReputationMiningCycle(true);
+      let repCycle = await IReputationMiningCycle.at(addr);
+      await forwardTime(MINING_CYCLE_DURATION + SUBMITTER_ONLY_WINDOW + 1, this);
+      await repCycle.submitRootHash("0x00", 0, "0x00", 10, { from: WORKER });
+      await repCycle.confirmNewHash(0, { from: MINER1 });
```

Comment:
Shouldn't this be `WORKER`?

Reply:
This is just setting up the test. It's the subsequent `delegatedClient` calls that test the functionality.